source/Calamari/ArgoCD/Helm/HelmValuesFileUpdateTargetParser.cs

```diff
@@ -3,195 +3,100 @@
+        readonly Dictionary<string, string> annotations;
+        readonly bool containsMultipleSources;
 
         public HelmValuesFileUpdateTargetParser(Application toUpdate, string defaultRegistry)
         {
-            aliasAnnotations = toUpdate.Metadata.Annotations
-                                            .Where(a => a.Key.StartsWith(ArgoCDConstants.Annotations.OctopusImageReplaceAliasKey))
-                                            .ToList();
-            imageReplacePathAnnotations = toUpdate.Metadata.Annotations
-                                                       .Where(a => a.Key.StartsWith(ArgoCDConstants.Annotations.OctopusImageReplacementPathsKey))
-                                                       .ToList();
+            annotations = toUpdate.Metadata.Annotations;
+            containsMultipleSources = toUpdate.Spec.Sources.Count > 1;
             helmSources = toUpdate.Spec.Sources.OfType<HelmSource>().ToList();
             refSources = toUpdate.Spec.Sources.OfType<ReferenceSource>().ToList();
             appName = toUpdate.Metadata.Name.ToApplicationName();
             this.defaultRegistry = defaultRegistry;
         }
 
-        public List<HelmValuesFileImageUpdateTarget> GetValuesFilesToUpdate()
+        public (IReadOnlyCollection<HelmValuesFileImageUpdateTarget> Targets, IReadOnlyCollection<HelmSourceConfigurationProblem> Problems) GetValuesFilesToUpdate()
         {
-            return helmSources
-                   .Where(hs => hs.Helm.ValueFiles.Count > 0)
-                   .SelectMany(ExtractValuesFilesForSource)
-                   .ToList();
+            var results = helmSources
+                              .Where(hs => hs.Helm.ValueFiles.Count > 0)
+                              .Select(ExtractValuesFilesForSource).ToArray();
+            
+            return (results.SelectMany(v => v.Targets).ToArray(), results.SelectMany(v => v.Problems).ToArray());
         }
 
-        List<HelmValuesFileImageUpdateTarget> ExtractValuesFilesForSource(HelmSource source)
+        (IReadOnlyCollection<HelmValuesFileImageUpdateTarget> Targets, IReadOnlyCollection<HelmSourceConfigurationProblem> Problems) ExtractValuesFilesForSource(HelmSource source)
         {
-            return source.Helm.ValueFiles.Select(file => file.StartsWith('$')
-                                                ? ProcessRefValuesFile(file)
-                                                : ProcessInlineValuesFile(source, file))
-                         .OfType<HelmValuesFileImageUpdateTarget>()
-                         .ToList();
+            var definedPathsForSource = ScopingAnnotationReader.GetImageReplacePathsForApplicationSource(source.Name.ToApplicationSourceName(), annotations, containsMultipleSources);
+            var results = source.Helm.ValueFiles.Select(file => file.StartsWith('$')
+                                                                ? ProcessRefValuesFile(file, definedPathsForSource, source)
```

Comment:
can we make processRefValuesFile and ProcessInlineValuesFile take their parameters in the same order?